src/daemon/plugin.c

```diff
@@ -1085,48 +1091,78 @@ static int plugin_compare_read_func(const void *arg0, const void *arg1) {
  * The AlignRead value is in seconds. The value modulo 60 determines the second
  * of a minute. The integer part of the value divided by 60 determines the
  * minute of an hour. Optionally, decimal places can be used to set the
- * fraction of a second (up to milliseconds precision).
+ * fraction of a second (up to microseconds precision).
  */
 static void plugin_set_first_read_time(read_func_t *rf) {
+  double alignRead = rf->rf_ctx.align_read; /* (default is -1.0) */
 
-  double start = rf->rf_ctx.align_read; /* (default is -1.0) */
-
-  /* if AlignRead is not set, start reading immediately */
-  if (start < 0) {
+  /* If AlignRead is not set, start reading immediately. */
+  if (alignRead < 0) {
     rf->rf_next_read = cdtime();
-  } else {
-    cdtime_t now = cdtime();
-
-    /* get minute in hour and second in minute and millisecond in second */
-    struct timeval tv = CDTIME_T_TO_TIMEVAL(now);
-    uint64_t curr_msec =
-        (((tv.tv_sec / 60) % 60) * 60 // minute of hour in seconds
-         + gmtime(&(tv.tv_sec))->tm_sec) *
-            1000             // add seconds of minute and convert to ms
-        + tv.tv_usec / 1000; // add millisecond of second
-
-    /* DEBUG("%s: current time: %lu sec, %lu ms in minute, start time: %.3lf,
-    interval: %lu", rf->rf_name, tv.tv_sec, curr_msec, start,
-    CDTIME_T_TO_MS(rf->rf_interval)); */
-
-    /* determine delay until next (aligned) read */
-    int64_t sleep_cdt = DOUBLE_TO_CDTIME_T(start) - MS_TO_CDTIME_T(curr_msec);
-    while (sleep_cdt < 0) {
-      sleep_cdt += rf->rf_interval;
+    return;
+  }
+
+  /* AlignRead values greater than Interval do not make sense. */
+  if (alignRead >= rf->rf_interval) {
+    static bool align_error = true;
+
+    if (align_error) {
+      ERROR("AlignRead has to be smaller than Interval (%.3lf >= %d)!",
+           alignRead, (int)rf->rf_interval);
```

Comment:
CDTIME_T_TO_DOUBLE(rf->rf_interval) should be used here.

Reply:
Done.